File: frontend/src/types/report.ts
import { AnalysisResult } from "./analysis";

export interface SavedReport {
  id: string;
  result: AnalysisResult;
  insight: string | null;
  created_at: string; // ISO-8601 UTC
}

export interface SaveReportRequest {
  result: AnalysisResult;
  insight?: string | null;
}

File: backend/app/models/report.py
from pydantic import BaseModel
from app.models.analysis import AnalysisResult


class SavedReport(BaseModel):
    id: str
    result: AnalysisResult
    insight: str | None
    created_at: str  # ISO-8601 UTC


class SaveReportRequest(BaseModel):
    result: AnalysisResult
    insight: str | None = None
